core/src/main/java/com/sequenceiq/distrox/v1/distrox/controller/DistroXV1Controller.java

```diff
@@ -85,15 +85,17 @@
 
     @Override
     public StackViewV4Responses list(String environmentName, String environmentCrn) {
-        StackViewV4Responses stackViewV4Responses;
-        if (!Strings.isNullOrEmpty(environmentName)) {
-            stackViewV4Responses =  stackOperations.listByEnvironmentName(
+        if (StringUtils.isNotEmpty(environmentName) && StringUtils.isNotEmpty(environmentCrn)) {
+            return stackOperations.listByEnvironmentCrn(
+                    workspaceService.getForCurrentUser().getId(), environmentCrn, StackType.WORKLOAD);
+        }
+        if (StringUtils.isNotEmpty(environmentName)) {
+            return stackOperations.listByEnvironmentName(
                     workspaceService.getForCurrentUser().getId(), environmentName, StackType.WORKLOAD);
         } else {
-            stackViewV4Responses = stackOperations.listByEnvironmentCrn(
+            return stackOperations.listByEnvironmentCrn(
                     workspaceService.getForCurrentUser().getId(), environmentCrn, StackType.WORKLOAD);
         }
```

Comment:
If environmentName is not empty but crn is empty, the `Line96` will call

Reply:
It is expected

Reply:
Please add unit test for these cases

Reply:
done